### ISSUE TITLE
created jenkinsfile refer to @master of library files

### ISSUE THREAD

**Issue by maxandersen** [openshiftio/openshift.io#123]
generated projects refer to `@Library('github.com/fabric8io/fabric8-pipeline-library@master')` rather than a specific version.

I would have expected a specific version or possibly a branch where only minor issues would be changing ?

**Comment by joshuawilson** [comment]
This issue was moved to fabric8io/fabric8-ui#1365

**Comment by maxandersen** [comment]
This issue still happens; just 2018-01-09 created a project in prod-preview and the same @master reference is inserted (see https://github.com/maxandersen/tonightshow2/blob/master/Jenkinsfile#L2)

The linked issue seem to just been closed without no update/context thus keeping this open.

**Comment by maxandersen** [comment]
p.s. pointing to @master wouldn't be an issue if we could guarantee at least some certainty the library would be backwards/forwards compatible in perpetuity.

**Comment by piyush1594** [comment]
@maxandersen Is this we need to workaround? We are going well with master branch :smiley:

**Comment by maxandersen** [comment]
is the head of master branch still able to build/run project created 6 months ago ? 

How about 12 months from now ? will master be able to build/run projects made today ?

**Comment by sthaha** [comment]
I am turning this into a feature request for the pipeline-library since what I gather from the issue is that it is asking the library to follow some consistent versioning system like semver.

**Comment by rupalibehera** [comment]
This looks like a feature request

**Comment by bartoszmajsak** [comment]
This requires a bit of automation too. All the jenkinsfiles are sourced from this repo https://github.com/fabric8io/fabric8-jenkinsfile-library, so whenever we have a new version and know it works (testing? ;)) - we should trigger update to this repo accordingly.

**Comment by lordofthejars** [comment]
@bartoszmajsak I have seen that all these things are already done, see https://github.com/fabric8io/fabric8-pipeline-library/blob/master/release.groovy#L19 so the question is if it has ever used (my bets are that it is since there are a lot of tags on repo). So the real question is why this was not used since 2017?

@sthaha @piyush1594 @rupalibehera @kishansagathiya Do you know anything about this? If there is still a job created for this somewhere?

**Comment by kishansagathiya** [comment]
@bartoszmajsak 
> so whenever we have a new version and know it works (testing? ;)) - we should trigger update to this repo accordingly.

Why is this required? I am under and impression that with a project we use a pipeline library version that is compatible with it. So, a project that was created 6 months ago would be creating a pipeline library version that was released six months ago. 
If we do this, below question would be answered in yes
> is the head of master branch still able to build/run project created 6 months ago ?

Isn't changing the project with the new pipeline version like using master?

**Comment by kishansagathiya** [comment]
@lordofthejars 
> Do you know anything about this? If there is still a job created for this somewhere?

Nope

**Comment by bartoszmajsak** [comment]
> Why is this required? I am under and impression that with a project we use a pipeline library version that is compatible with it. So, a project that was created 6 months ago would be creating a pipeline library version that was released six months ago.

@kishansagathiya not sure if we are talking about the same thing so let me rephrase.

When you create new project (or import to OSIO) we get Jenkinsfiles from the https://github.com/fabric8io/fabric8-jenkinsfile-library - they are all pointing to `master` https://github.com/fabric8io/fabric8-pipeline-library. If we start making "releases" those files should be updated accordingly, otherwise we don't fix anything from OSIO standpoint.

**Comment by chmouel** [comment]
This is going to be addresse with osio-pipeline library as we plan to do versionning properly there,

can you confirm @sthaha or @hrishin and closes this year and half old issue if that's correct,

**Comment by hrishin** [comment]
Yes @chmouel  that's the plan with osio-pipeline library .
https://github.com/fabric8io/osio-pipeline/issues/72

**Comment by chmouel** [comment]
perfect closed ;)
